server/src/services/hcs-ballot-processing.service.ts

```diff
@@ -96,38 +94,53 @@ export class HcsBallotProcessingService {
 		} else if (ballot.ineligibleAccounts.find((a) => !is_entity_id(a))) {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Invalid account id found in Ineligible account list.`);
 		} else if (-1 !== ballot.ineligibleAccounts.findIndex((a) => !is_entity_id(a))) {
 			this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Found invalid account id in list of inelegible accounts.`);
 		} else {
-			ballot.ineligibleAccounts = [...new Set([...ballot.ineligibleAccounts, ...this.config.ineligibleAccounts])];
 			return async () => {
-				if (await this.dataService.getBallot(ballot.consensusTimestamp)) {
+				const rule = this.dataService.getLatestRule(hcsMirrorRecord.consensus_timestamp as unknown as TimestampKeyString);
+
+				if (!rule) {
+					this.logger.verbose(
+						`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: No rule found for ballot. Make sure a define-rules message is submitted before creating a ballot.`,
+					);
+				} else if (rule.ballotCreators?.length > 0 && rule.ballotCreators.indexOf(ballot.payerId) === -1) {
+					this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Payer ID is not on the allowed list.`);
+				} else if (await this.dataService.getBallot(ballot.consensusTimestamp)) {
 					this.logger.verbose(`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Ballot with same consensus timestamp already exists.`);
+				} else if (ballot.minVotingThreshold < rule.minVotingThreshold) {
+					this.logger.verbose(
+						`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Invalid Minimum Voting Threshold, must be equal to or greater than global configuration of ${rule.minVotingThreshold}.`,
+					);
+				} else if (computeDiffInDays(ballot.consensusTimestamp, ballot.startTimestamp) < rule.minimumStandoffPeriod) {
+					this.logger.verbose(
+						`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Voting starting time is less than ${rule.minimumStandoffPeriod} days from ballot creation.`,
+					);
+				} else if (computeDiffInDays(ballot.startTimestamp, ballot.endTimestamp) < rule.minVotingThreshold) {
+					this.logger.verbose(
+						`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: Voting ending time does not permit required voting windo of ${rule.minVotingThreshold} days.`,
+					);
+				} else if (ballot.tokenId !== rule.tokenId) {
+					this.logger.verbose(
+						`Message ${hcsMessage.sequenceNumber} failed create-ballot validation: contains a create-ballot for a different payment token ID.`,
+					);
 				} else {
+					ballot.ineligibleAccounts = [...new Set([...ballot.ineligibleAccounts, ...(rule.ineligibleAccounts ? rule.ineligibleAccounts : [])])];
+					ballot.minVotingThreshold = ballot.minVotingThreshold || rule.minVotingThreshold || 0;
+					ballot.hcsToken = rule.hcsToken;
```

Comment:
This new property is added in order to keep track of the token used for this ballot since the token could be changed in a future rule so we need to keep a local reference to the ballot.